eq-author/src/App/QuestionnairesPage/QuestionnairesView/QuestionnairesTable/index.js

```diff
@@ -26,43 +30,121 @@ const TH = styled.th`
   font-size: 0.9em;
 `;
 
-TH.propTypes = {
-  colWidth: PropTypes.string.isRequired,
-};
+const SortButton = styled.span`
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  opacity: 0.8;
+
+  &::after {
+    content: url(${iconArrow});
+    display: inline-block;
+    width: 16px;
+    height: 16px;
+    opacity: ${props => (props.active ? "0.8" : "0.2")};
+    transform: rotate(${props => (props.order === DESC ? "0deg" : "180deg")});
+  }
+
+  &:hover {
+    opacity: 1;
+  }
+`;
 
 const Panel = styled.div`
   background: white;
   border-radius: 4px;
   border: 1px solid #ccc;
 `;
 
+const SortableTH = ({
+  children,
+  sortOrder,
+  sortKey,
+  onSortClick,
+  onReverseClick,
+  colWidth,
+  currentSortKey,
+  dataTest,
+}) => {
+  const active = sortKey === currentSortKey;
+  return (
+    <TH aria-sort={active ? sortOrder : "none"} colWidth={colWidth}>
+      <SortButton
+        active={active}
+        order={active ? sortOrder : DESC}
+        role="button"
+        onClick={() => (active ? onReverseClick() : onSortClick(sortKey))}
+        data-test={dataTest}
+      >
+        {children}
+      </SortButton>
+    </TH>
+  );
+};
+
+const TableHead = props => {
+  return (
+    <thead>
+      <tr>
+        <SortableTH
+          sortKey="title"
+          colWidth="50%"
+          dataTest="title-sort-button"
+          {...props}
+        >
+          Title
+        </SortableTH>
+        <SortableTH sortKey="createdAt" colWidth="15%" {...props}>
+          Created
+        </SortableTH>
+        <SortableTH sortKey="createdBy.id" colWidth="20%" {...props}>
+          Owner
+        </SortableTH>
+        <TH colWidth="15%">Actions</TH>
+      </tr>
+    </thead>
+  );
+};
+
 const QuestionnairesTable = ({
   questionnaires,
   onDeleteQuestionnaire,
   onDuplicateQuestionnaire,
+  onSortQuestionnaires,
+  onReverseSort,
   autoFocusId,
+  sortColumn,
+  sortOrder,
 }) => {
+  let displayedQuestionnaires = questionnaires;
+  displayedQuestionnaires = sortBy(displayedQuestionnaires, questionnaires =>
+    !sortColumn ? questionnaires : get(questionnaires, sortColumn).toLowerCase()
+  );
```

Comment:
This will show the oldest ones at the top. So, if you sorted by title descending and some of them have the same name, the oldest ones will appear at the top.

Reply:
Ah good point, in that case just doing `sortBy(questionnaires, [sortColumn.toLowerCase()])` should work as lodash preserves the original sort order.

Reply:
```js
sortBy(questionnaires, [
        questionnaire => questionnaire[sortColumn].toUpperCase(),
        "createdAt",
      ])
```